### DISCUSSION TITLE
Consistent UI/UX for Choosing Document Types

### DISCUSSION THREAD

**Discussion by markadrake** [umbraco/Umbraco-CMS#12402] [Features and ideas]
The dialog used to display document types (when building a composition) has been well thought out and offers a superior user experience to other dialogs I've seen. Here is a screen capture:

<img width="500" alt="image" src="https://user-images.githubusercontent.com/713355/168452059-2cf6e7ee-eb46-41a3-a84b-3ad3eb6621d7.png">

The Block List property editor also makes use of a dialog. This dialog has fewer options – you cannot search, and traversing the tree becomes quite repetitive. Here is a screen capture:

<img width="501" alt="image" src="https://user-images.githubusercontent.com/713355/168452107-b4f1f189-7568-4b51-a8df-50efca3bbe02.png">

Would there be positive reception to a PR for replacing the above dialog with the more comprehensive one? 

I understand that there is a workflow for the block list editor. This workflow, however, has no required fields. I would propose that if a single element type is selected, we can show the follow-on dialog. If multiple elements are selected, we use the default configuration and do not show the follow-on dialog.

All the best,

**Comment by hfloyd** [comment]
I like your idea, @markadrake.
I wonder if the inconsistency is partly due to a 'single' picker versus a 'multiple' picker (checkboxes)?
